### PR TITLE
CYLC_SITE_CONF_PATH: look in flow sub-dir

### PR DESCRIPTION
**Pull request by oliver-sanders**
Required for: https://github.com/cylc/cylc-uiserver/pull/192

Required for us to use `CYLC_SITE_CONF_PATH` to source the global config for the UIS.

* Look for global config in `${CYLC_SITE_CONF_DIR}/flow/global.cylc`.
* This is to allow this env var to be used by other cylc components - namely the UIS.
* Tried to match the annotations from #4099 to minimise diff.

To explain the `:envvar:` stuff - https://github.com/cylc/cylc-doc/issues/226

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (why?).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] Docs included
- [x] No dependency changes.

### CODE REVIEW

**Comment by hjoliver** [comment]
(The new unit test is failing)

**Comment by oliver-sanders** [comment]
That should do it, nicer testing approach which bypasses the FS all-together.

File: cylc/flow/cfgspec/globalcfg.py [hunk 1]
```diff
@@ -59,24 +59,48 @@
...
     from those higher up (but if a setting is present in a file higher up and
     not in any files lower down, it will not be overridden).
 
-    Setting the ``CYLC_SITE_CONF_PATH`` environment variable overrides the
-    default value of ``/etc/cylc/flow/``.
+    The following environment variables can change the files which are loaded:
+
+    .. envvar:: CYLC_CONF_PATH
+
+       If set this bypasses the default site/user configuration hierarchy used
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:84]:
```suggestion
       If set, this bypasses the default site/user configuration hierarchy used
```

**Review by datamel — APPROVED** [review]
I have read the code. Built/checked the docs and run the tests. I'm happy with this change.

File: cylc/flow/cfgspec/globalcfg.py [hunk 2]
```diff
@@ -59,24 +59,48 @@
...
+
+       This should be set to a directory containing a :cylc:conf:`global.cylc`
+       file.
+
+    .. envvar:: CYLC_SITE_CONF_PATH
 
-    To override the entire hierarchy, set the ``CYLC_CONF_PATH`` environment
-    variable to the directory containing your ``global.cylc`` file.
+       By default the site configuration is located in ``/etc/cylc/``. For
+       installations where this is not convenient this path can be overridden
```
Comment (MetRonnie) [inline cylc/flow/cfgspec/globalcfg.py:93]:
```suggestion
       installations where this is not convenient, this path can be overridden
```

File: tests/unit/scripts/test_config.py [hunk 3]
```diff
@@ -14,31 +14,186 @@
...
+from cylc.flow.cfgspec.globalcfg import GlobalConfig
+
 
 Fixture = Any
+HOME: str = str(Path('~').expanduser())
+
+
+@pytest.fixture
+def conf_env(monkeypatch):
+    """Clear any env vars that effect which conf files get loaded.
```
Comment (MetRonnie) [inline tests/unit/scripts/test_config.py:33]:
```suggestion
    """Clear any env vars that affect which conf files get loaded.
```

File: tests/unit/scripts/test_config.py [hunk 4]
```diff
@@ -14,31 +14,186 @@
...
+    )
+    return files
+
+
+def test_get_config_file_hierarchy_global(
+    monkeypatch: Fixture,
+    conf_env: Fixture,
+    capload: Fixture,
+    dummy_version_hierarchy: Fixture
+):
```
Comment (MetRonnie) [inline tests/unit/scripts/test_config.py:96]: I didn't realise just by passing fixtures to test functions that they would be executed

**Review by MetRonnie — APPROVED** [review]
Built docs and ran tests. Just a few optional suggestions

File: tests/unit/scripts/test_config.py (hunk 4 above)
Comment (oliver-sanders) [reply tests/unit/scripts/test_config.py:96]: Up until the `return` or `yield` statement, for generators execution continues after the `yield` statement after the test scope exits.

**Pull request merged**